app/src/main/java/ch/logixisland/anuto/view/faq/EnemiesAdapter.java

```diff
@@ -0,0 +1,212 @@
+package ch.logixisland.anuto.view.faq;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Matrix;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.lang.ref.WeakReference;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import ch.logixisland.anuto.R;
+import ch.logixisland.anuto.engine.theme.Theme;
+import ch.logixisland.anuto.entity.enemy.EnemyProperties;
+import ch.logixisland.anuto.entity.enemy.EnemyType;
+
+public class EnemiesAdapter extends BaseAdapter {
+    private static String sTheme;
+    private static Map<EnemyType, Bitmap> sEnemyCache;
+
+    private final List<EnemyProperties> mEnemyProperties;
+
+    private final WeakReference<Activity> mActivityRef;
+    private Context appContext;
+    private Theme mTheme;
+
+    public EnemiesAdapter(Activity activity, Context context, Theme theme) {
+        mActivityRef = new WeakReference<>(activity);
+        appContext = context;
+        mTheme = theme;
+
+        if ((sTheme != theme.getName()) || (sEnemyCache == null)) {
+            sTheme = theme.getName();
+            sEnemyCache = new HashMap<>();
+        }
+
+        mEnemyProperties = new ArrayList<>();
+        for (EnemyType x : EnemyType.values()) {
+            mEnemyProperties.add(new EnemyProperties.Builder(x.name()).build());
+        }
+    }
+
+    static private class ViewHolder {
+        ImageView img_enemy;
+        TextView txt_name;
+        TextView txt_health;
+        TextView txt_speed;
+        TextView txt_reward;
+        TextView txt_weak_against;
+        TextView txt_strong_against;
+
+        ViewHolder(View view) {
+            img_enemy = view.findViewById(R.id.img_enemy);
+            txt_name = view.findViewById(R.id.txt_name);
+            txt_health = view.findViewById(R.id.txt_health);
+            txt_speed = view.findViewById(R.id.txt_speed);
+            txt_reward = view.findViewById(R.id.txt_reward);
+            txt_weak_against = view.findViewById(R.id.txt_weak_against);
+            txt_strong_against = view.findViewById(R.id.txt_strong_against);
+        }
+    }
+
+    @Override
+    public int getCount() {
+        return mEnemyProperties.size();
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return mEnemyProperties.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return 0;
+    }
+
+    private Bitmap extractSingleBmp(EnemyType enemyType) {
+        int attrId, spriteCount, spriteId;
+
+        switch (enemyType) {
```

Comment:
Same argumentation as for the EnemyProperties. Would be better to get the bitmaps directly from the Enemy classes using the EntityRegistry.

Reply:
Sure, but won't that be to much of an hassle?